kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java

```diff
@@ -220,32 +229,71 @@ public KyuubiConnection(String uri, Properties info) throws SQLException {
     client = newSynchronizedClient(client);
   }
 
-  private void getLaunchEngineLog() {
+  /**
+   * Check whether launch engine operation might be producing more logs to be fetched.
+   * This method is a public API for usage outside of Kyuubi, although it is not part of the
+   * interface java.sql.Connection.
+   * @return true if launch engine operation might be producing more logs. It does not indicate
+   *         if last log lines have been fetched by getEngineLog.
+   */
+  public boolean hasMoreEngineLogs() {
+    return launchEngineOpHandle != null && !(launchEngineOpCompleted && !isEngineLogBeingGenerated);
```

Comment:
Maybe write as `launchEngineOpHandle != null && (isEngineLogBeingGenerated || !launchEngineOpCompleted)` is more readable?